subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/GroovySourceIncrementalCompilationIntegrationTest.groovy

```diff
@@ -182,4 +182,24 @@ class A2{}
         !file('build/tmp/compileGroovy/source-classes-mapping.txt').text.contains('A.groovy')
         file('build/tmp/compileGroovy/source-classes-mapping.txt').text.contains('B.groovy')
     }
+
+    def "does recompile when a resource changes"() {
```

Comment:
Why do we recompile here?

Reply:
Any changes to non-Groovy files cause a full recompile for Groovy, since we don't know which classes those resources affect. Note that this is only for resources in the Groovy source directory.

Reply:
Is this about properties files registering transforms and the like? Can we be more specific about which resources cause full recompile (now or later)?

Reply:
Currently, any file which is not `.groovy` causes a full recompile.

In the future, I guess changes to the file registering global AST transform also need to cause a full rebuild. An AST transform can read any resource file I guess, though I don't know. Then any resource change (so non-Java and non-Groovy) will cause a recompilation.

I am not 100% sure about it though. I will add the above comment to the test. WDYT?

Reply:
👍